tests/test_bindings.py

```diff
@@ -1,33 +1,47 @@
+'''TESTS BINDING'''
+
+import os
 import json
 import yaml
-import vttv
-
 import sys
-import os
+
+import vttv
 
 # source dir is the directory a level above this file
 source_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
+# Read the YAML config file
 with open(f'{source_dir}/tests/test_bindings_conf.yaml', 'r') as stream:
-  try:
-    params = yaml.safe_load(stream)
-  except yaml.YAMLError as exc:
-    print(exc)
+    try:
+        params = yaml.safe_load(stream)
+    except yaml.YAMLError as exc:
+        print(exc)
+
+# Check main key is "visualization"
+if 'visualization' not in params:
+    print("The YAML configuration file is not valid: missing required paramaters: visualization")
+    sys.exit(1)
 
 # make output_visualization_dir directory parameter absolute
-params["visualization"]["output_visualization_dir"] = os.path.abspath(params["visualization"]["output_visualization_dir"])
+if 'output_visualization_dir' in params["visualization"]:
+    params["visualization"]["output_visualization_dir"] = os.path.abspath(
+      params["visualization"]["output_visualization_dir"]
+    )
 
 params_serialized = yaml.dump(params["visualization"])
 
-n_ranks = params["visualization"]["x_ranks"] * params["visualization"]["y_ranks"] * params["visualization"]["z_ranks"]
-rank_data = []
+n_ranks = params["visualization"]["x_ranks"]
```

Comment:
What is the reason for separating this in 3 lines, requiring 2 additional assignments?

Reply:
Pylint warning: Line too long